src/main/java/io/numaproj/numaflow/sessionreducer/Service.java

```diff
@@ -79,7 +84,29 @@ public StreamObserver<Sessionreduce.SessionReduceRequest> sessionReduceFn(final
             public void onNext(Sessionreduce.SessionReduceRequest sessionReduceRequest) {
                 // send the message to parent actor, which takes care of distribution.
                 if (!supervisorActor.isTerminated()) {
-                    supervisorActor.tell(sessionReduceRequest, ActorRef.noSender());
+                    // if the operation is a MERGE, make it a blocking call.
+                    if (sessionReduceRequest.getOperation().getEvent()
+                            == Sessionreduce.SessionReduceRequest.WindowOperation.Event.MERGE) {
+                        // set time out to 1 second as we expect a MERGE operation to finish quickly.
+                        Timeout timeout = new Timeout(Duration.create(1, "seconds"));
+                        try {
```

Comment:
Hacky, sometimes it might take more than 1 second

Reply:
thanks, what should be the proper number? @yhl25

Reply:
let's wait until the it returns

Reply:
also, please add unit tests to cover merge followed by another operation to make sure the blocking call works!

Reply:
The method Await asks for a timeout, I don't think we should wait forever, right? Maybe we should set an atMost timeout value.

Yes, the uts for merge are already in place. Please check the serverTest class.